Lib/zipimport.py

```diff
@@ -715,4 +712,4 @@ def _get_module_code(self, fullname):
             modpath = toc_entry[0]
             return code, ispackage, modpath
     else:
-        raise ZipImportError(f"can't find module {fullname!r}", name=fullname)
+        raise ZipImportError(error_msg or f"can't find module {fullname!r}", name=fullname)
```

Comment:
consider whether we should be replacing the error message here or if we should use Python's exception chaining and have this be something like:

```python
if import_error:  # the exception `e` saved above instead of `error_msg`
    raise ZipImportError(f"can't load module {fullname!r}", name=fullname) from import_error
else:
    raise ZipImportError(f"can't find module {fullname!r}", name=fullname)

Reply:
I considered that, but wasn't sure this adds more information on top of the original exception. On the other hand I don't like copying the message from one exception to the other, so definitely open to it.

Reply:
I would vote for chaining.

Reply:
and if you want the message in the final exception to be more directly helpful, you can modify it as well even when chaining. :)

the chaining is primarily useful for debugging purposes to figure out exactly where and why the underlying cause occurred.  the top level exception's message is the one most likely to find its way to a user or developer bugreport rather than necessarily the combination of chained tracebacks.